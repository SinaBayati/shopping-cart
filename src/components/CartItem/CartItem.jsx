import styled from "styled-components";

const Wrapper = styled.div`
  padding: 2rem;
  border-radius: 0.5rem;
  background-color: black;
  box-shadow: 0 0 10px black;
  color: white;
  font-size: 1.2rem;
  font-family: "Gill Sans", "Gill Sans MT", Calibri, "Trebuchet MS", sans-serif;
  display: flex;
  align-items: center;
  gap: 4rem;

  img {
    width: 100px;
    height: auto;
    border-radius: 0.5rem;
  }
`;

const FlexWrapper = styled.div`
  display: flex;
  flex-direction: column;
  gap: 1rem;
`;

const StyledInput = styled.input`
  width: 4rem;
  background-color: #111;
  color: white;
  border: none;
  border-radius: 0.5rem;
  padding: 0.5rem;
  font-size: 1rem;
`;

const StyledButtonSmall = styled.button`
  padding: 0.3rem 0.5rem;
  border-radius: 0.5rem;
  font-size: 1.2rem;
  background-color: #111;
  color: white;
  border: none;
  cursor: pointer;

  &:hover {
    box-shadow: 0 0 5px white;
  }
`;

const ButtonsWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 0.5rem;
`;

const RemoveButton = styled(StyledButtonSmall)`
  padding: 0.5rem 2rem;
`;

function CartItem({
  id,
  title = "Cart Item",
  count = 0,
  price = "???$",
  onIncrement,
  onDecrement,
  onCountChange,
  onRemove,
  imgSrc = "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
  imgAlt,
}) {
  return (
    <Wrapper>
      <img src={imgSrc} alt={imgAlt} />
      <FlexWrapper>
        <h3>{title}</h3>
        <h4>{price + "$"}</h4>
        <ButtonsWrapper>
          <StyledButtonSmall onClick={() => onDecrement(id)}>
            -
          </StyledButtonSmall>
          <StyledInput
            value={count}
            onChange={(e) => onCountChange({ id, value: e.target.value })}
            type="number"
            min={0}
          ></StyledInput>
          <StyledButtonSmall onClick={() => onIncrement(id)}>
            +
          </StyledButtonSmall>
        </ButtonsWrapper>
      </FlexWrapper>
      <RemoveButton onClick={() => onRemove(id)}>Remove Item</RemoveButton>
    </Wrapper>
  );
}

export default CartItem;
