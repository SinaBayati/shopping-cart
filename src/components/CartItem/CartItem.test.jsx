import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import CartItem from "./CartItem";

describe("CartItem", () => {
  it("renders CartItem correctly", () => {
    const { container } = render(<CartItem />);

    expect(container).toMatchSnapshot();
  });
});
