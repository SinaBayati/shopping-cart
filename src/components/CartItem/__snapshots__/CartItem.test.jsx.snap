// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CartItem > renders CartItem correctly 1`] = `
<div>
  <div
    class="sc-Qotzb gdaYHo"
  >
    <img
      src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
    />
    <div
      class="sc-fYsHOw hbMOVr"
    >
      <h3>
        Cart Item
      </h3>
      <h4>
        ???$$
      </h4>
      <div
        class="sc-dIMoHT inWsHY"
      >
        <button
          class="sc-fWnslK ilLYJC"
        >
          -
        </button>
        <input
          class="sc-dKREkF fVVDmn"
          min="0"
          type="number"
          value="0"
        />
        <button
          class="sc-fWnslK ilLYJC"
        >
          +
        </button>
      </div>
    </div>
    <button
      class="sc-fWnslK sc-iQQCXo ilLYJC eBkRtH"
    >
      Remove Item
    </button>
  </div>
</div>
`;
